### PR TITLE
0 port support

### PR DESCRIPTION
**Pull request by kemphack**

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -93,7 +93,10 @@ export class Server extends EventEmitter {
 
         options = options || {};
 
-        this.port = options.port || DEFAULT_PROXY_SERVER_PORT;
+        if (options.port === undefined)
```
Comment (jancurn) [inline src/server.js:96]: Please can you also test for `null` ?

File: src/server.js [hunk 2]
```diff
@@ -449,6 +452,7 @@ export class Server extends EventEmitter {
                 reject(err);
             };
             const onListening = () => {
+                this.port = this.server.address().port;
```
Comment (jancurn) [inline src/server.js:455]: Can we update some existing unit test to check that `this.port` remains set correctly to the previous value, if it was null, undefined or 0?

**Review by jancurn — CHANGES_REQUESTED** [review]
Thanks for the PR, just two small comments, otherwise it's good to go.

File: src/server.js (hunk 2 above)
Comment (kemphack) [reply src/server.js:455]: not sure, i added separate test for the first time

**Review by jancurn — APPROVED** [review]
